runbotics/runbotics-scheduler/src/queue/queue.service.ts

```diff
@@ -90,8 +91,33 @@ export class QueueService implements OnModuleInit {
                 );
                 return Promise.reject(err);
             });
+        if (input?.timeout) {
```

Comment:
If timeout provided, set the timeout callback function which checks if job is free to be removed and then removes the job and add timer to map of timers